lib/gstreamer/d3d12/decoder.py

```diff
@@ -0,0 +1,97 @@
+###
+### Copyright (C) 2024 Intel Corporation
+###
+### SPDX-License-Identifier: BSD-3-Clause
+###
+
+import os
+import slash
+
+from ....lib import platform
+from ....lib.codecs import Codec
+from ....lib.common import get_media
+from ....lib.formats import PixelFormat
+from ....lib.gstreamer.decoderbase import BaseDecoderTest, Decoder as GstDecoder
+from ....lib.gstreamer.util import have_gst_element
+from ....lib.gstreamer.d3d12.util import mapformat, mapformatu
+
+class Decoder(GstDecoder):
+  format  = property(lambda s: mapformatu(super().format))
+  pformat = property(lambda s: mapformat(super().format))
+
+@slash.requires(*have_gst_element("d3d12"))
+class DecoderTest(BaseDecoderTest):
+  DecoderClass = Decoder
+
+  def before(self):
+    super().before()
+
+  def get_supported_format_map(self):
+    return {
+      "I420"  : "yuv420p",
+      "NV12"  : "nv12",
+      "P010"  : "p010le",
+    }
```

Comment:
This member function never gets called anywhere because none of the gstreamer decoder classes inherit the FormatMapper/BaseFormatMapper classes.  Only the ffmpeg classes have implemented this so far.  The gstreamer classes have not been updated for it, yet.  Currently, gstreamer code still uses the top-level util method (non-member).